src/main/java/mate/academy/bookshop/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.bookshop.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.shoppingcart.CartItemDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.CartItemMapper;
+import mate.academy.bookshop.model.Book;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.repository.CartItemRepository;
+import mate.academy.bookshop.service.CartItemService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public CartItemDto updateById(Long id, int quantity) {
+        CartItem cartItemById = getById(id);
+        cartItemById.setQuantity(quantity);
+        return cartItemMapper.toDto(cartItemRepository.save(cartItemById));
+    }
+
+    @Override
+    public void deleteCartItem(Long id) {
```

Comment:
it is CartItem service so you can just call this method "deleteById"

Reply:
agree :)